### PR TITLE
native: change Dlahqr back to return single integer

### PR DESCRIPTION
**Pull request by vladimir-ch**
If the returned integer is a 1-based size and not a 0-based index, we can use 0 to indicate success (i.e., all eigenvalues have converged) and thus the `converged bool` is not necessary.

Also polished the docs a little.

PTAL

### CODE REVIEW

File: native/dlahqr.go [hunk 1]
```diff
@@ -37,37 +36,34 @@ import (
...
-// If converged is false and wantt is true, then on return
+// If unconverged is positive, some eigenvalues have not converged, and
+// wr[unconverged:ihi+1] and wi[unconverged:ihi+1] contain those eigenvalues
+// which have been successfully computed. If wantt is false, on return the
+// remaining unconverged eigenvalues are the eigenvalues of the upper Hessenberg
+// matrix H[ilo:unconverged,ilo:unconverged]. If wantt is true, then on return
 //  (initial H)*U = U*(final H),   (*)
 // where U is an orthogonal matrix. The final H is upper Hessenberg and
-// H[index+1:ihi+1,index+1:ihi+1] is upper triangular.
+// H[unconverged:ihi+1,unconverged:ihi+1] is upper quasi-triangular.
```
Comment (vladimir-ch) [inline native/dlahqr.go]: I think that reference LAPACK means _quasi_-triangular here, triangular does not make much sense.

File: native/dlahqr.go [hunk 2]
```diff
@@ -25,7 +24,7 @@ import (
 //  H[ilo,ilo-1] == 0,  if ilo > 0,
 // otherwise Dlahqr will panic.
 //
-// wr[ilo:ihi+1] and wi[ilo:ihi+1] will contain, on return if converged is true,
+// If unconverged is zero, wr[ilo:ihi+1] and wi[ilo:ihi+1] will contain on return
```
Comment (kortschak) [inline native/dlahqr.go]:
```
// If unconverged is zero on return, wr[ilo:ihi+1] and wi[ilo:ihi+1] will contain
// respectively the real and imaginary parts of the computed eigenvalues ilo
// to ihi.
```

Keep the parse tree a little less branched.

**Comment by kortschak** [comment]
LGTM with minor edit.

File: native/dlahqr.go (hunk 2 above)
Comment (vladimir-ch) [inline native/dlahqr.go]: Ok, returned to the previous style.

Comment (vladimir-ch) [inline native/dlahqr.go]: Ah, reading the comment in email I missed your suggested edit and misunderstood the message. Will send as a separate PR.
